rust/kernel/sync/arc.rs

```diff
@@ -7,20 +7,24 @@
 //! underlying object when it reaches zero. It is also safe to use concurrently from multiple
 //! threads.
 //!
-//! It is different from the standard library's [`Arc`] in two ways: it does not support weak
-//! references, which allows it to be smaller -- a single pointer-sized integer; it allows users to
-//! safely increment the reference count from a single reference to the underlying object.
+//! It is different from the standard library's [`Arc`] in at least three ways:
+//! 1. It does not support weak references, which allows it to be smaller: a single pointer-sized
+//!    integer;
+//! 2. It allows users to safely increment the reference count from a single reference to the
+//!    underlying object. That is, go from `&T` to [`Ref<T>`];
+//! 3. It saturates the reference count instead of aborting when it goes over a threshold.
 //!
 //! [`Arc`]: https://doc.rust-lang.org/std/sync/struct.Arc.html
 
-use crate::Result;
+use crate::{bindings, Result};
 use alloc::boxed::Box;
-use core::{
-    mem::ManuallyDrop,
-    ops::Deref,
-    ptr::NonNull,
-    sync::atomic::{fence, AtomicUsize, Ordering},
-};
+use core::{cell::UnsafeCell, convert::AsRef, mem::ManuallyDrop, ops::Deref, ptr::NonNull};
+
+extern "C" {
+    fn rust_helper_refcount_set(r: *mut bindings::refcount_t, n: u32);
```

Comment:
Should `n: u32` be `n: c_types::c_int` instead?
https://github.com/Rust-for-Linux/linux/blob/d8c6b9ce7fb78375460422ba15d763c22f70d0af/include/linux/refcount.h#L129-L137

Reply:
Indeed. Thanks for catching this, I was looking at an earlier version that used `unsigned int` (https://elixir.bootlin.com/linux/v5.4.125/source/include/linux/refcount.h#L30).